### PR TITLE
Add rhonewsn1 to nuopc and to the NOAH MP driver and source code

### PR DESCRIPTION
**Pull request by ericaligo-NOAA**
A variable precip ice density, rhonewsn1, is passed into the driver and then assigned to prcpiceden. The array prcpiceden is used in the LSM if the namelist option, iopt_snf = 5 and exticeden = T. The flag exticeden is set to T if a user wants to compute the variable precip ice density.

This PR is connected with the following PRs:

Depends on Winterwx NOAA-EMC/fv3atm#614
Depends on Winterwx ufs-community/ccpp-physics#30
Six new winter weather diagnostics, new precip ice density for RUC, NOAH and NOAH MP LSM, and rename nsradar_reset ufs-community/ufs-weather-model#1529

### CODE REVIEW

**Comment by ChunxiZhang-NOAA** [comment]
@barlage @uturuncoglu Could you review this PR at your convenience?

**Comment by uturuncoglu** [comment]
@ChunxiZhang-NOAA I have a meeting all day but I'll look at it today and maybe test it.

**Comment by ChunxiZhang-NOAA** [comment]
@uturuncoglu No problem. Thank you!

File: drivers/nuopc/lnd_comp_driver.F90
Comment (uturuncoglu) [inline drivers/nuopc/lnd_comp_driver.F90:577]: I am not sure how the new variable handled in CCPP/Physics side but I think the cap also need to provide it to be consistent with the CCPP version of the Noah-MP. As I understand from you implementation, the variable is initialized to 0 in component version (NUOPC cap) and does not change but I think in the CCPP/Physics implementation it is chnaging. Since we are comparing component version with the CCPP/Physics version of the Noah-MP this will create difference between them. Is there any specific code in CCPP/Physics side that calculates this new argument. If so, that needs to be ported to NUOCP cap too. Is there any way to calculate it also in NUOPC cap?

Comment (uturuncoglu) [reply drivers/nuopc/lnd_comp_driver.F90:577]: I think it is calculated under `GFS_MP_generic_post.F90` but not sure how same can be done in component model. @barlage do you have any idea about it? If CCPP/Physics provide it but component version does not then this means that the results will be different in both side.

**Review by uturuncoglu — CHANGES_REQUESTED** [review]
We need to clarify how we handle this new argument in component version of the model before merging this. So, we also need input from @barlage

File: drivers/nuopc/lnd_comp_driver.F90
Comment (barlage) [reply drivers/nuopc/lnd_comp_driver.F90:577]: In my understanding, the short term solution is keep this as zero in the component version, this is only needed for one opt_snf option; eventually we will want to bring this through the cap. I don't think that we want to calculate this in the component model or land cap since we will need to pass several fields into the component model to calculate this.

Comment (uturuncoglu) [reply drivers/nuopc/lnd_comp_driver.F90:577]: @barlage Then, we could merge it as it is. Right?

Comment (barlage) [reply drivers/nuopc/lnd_comp_driver.F90:577]: Yes.

Comment (barlage) [reply drivers/nuopc/lnd_comp_driver.F90:577]: As long as we don't activate that option in the tests, the results should be identical.

**Review by uturuncoglu — APPROVED** [review]

**Comment by uturuncoglu** [comment]
@ChunxiZhang-NOAA @BrianCurtis-NOAA This is merged now. Thanks for putting this.

**Comment by ChunxiZhang-NOAA** [comment]
@uturuncoglu Thanks for merging it. The merge is supposed to be done AFTER all ufs-weather-model regression tests on different platforms passed. But I think it is OK for now unless some bugs caught later from the regression tests.

**Comment by uturuncoglu** [comment]
@ChunxiZhang-NOAA I was thinking that that is already done and you are waiting for this merge. Anyway, if you see any issue we could fix it. I have already test the datm+land outside of the UFS and it was fine. I am not sure about other RTs.

**Comment by ChunxiZhang-NOAA** [comment]
@uturuncoglu Sure, I will let you know if there is any issue. And yes, the RT test on Hera was successful. The chance to fail on other platforms is very low.

**Pull request merged**
